test/common/http/conn_pool_grid_test.cc

```diff
@@ -40,21 +40,28 @@ class ConnectivityGridForTest : public ConnectivityGrid {
     return grid.getHttp3StatusTracker().hasHttp3FailedRecently();
   }
 
-  static ConnectionPool::Instance* forceCreateNextPool(ConnectivityGrid& grid) {
-    return grid.createNextPool();
+  static ConnectionPool::Instance* forceGetOrCreateHttp3Pool(ConnectivityGrid& grid) {
+    return grid.getOrCreateHttp3Pool();
+  }
+  static ConnectionPool::Instance* forceGetOrCreateHttp2Pool(ConnectivityGrid& grid) {
+    return grid.getOrCreateHttp2Pool();
   }
 
-  ConnectionPool::Instance* createNextPool() override {
-    if (http2_pool_ && http3_pool_) {
-      return nullptr;
+  ConnectionPool::Instance* getOrCreateHttp3Pool() override {
+    if (!http3_pool_) {
+      http3_pool_.reset(createMockPool("http3"));
```

Comment:
nit: `http3_pool_ = createMockPool("http3")` (ditto below) or alternatively if we go with making `createFooPool()` methods, then this is even simpler.

Reply:
replaced